test/hardwareWallet.js

```diff
@@ -0,0 +1,59 @@
+const { assert } = require('chai');
+let bcoin;
+
+if (process.env.BROWSER) bcoin = require('bcoin/lib/bcoin-browser');
+else bcoin = require('bcoin');
+
+const { HDPublicKey } = bcoin;
+const { HardwareWallet } = require('../lib/blockchain');
+
+describe('ledger hardware wallet', async () => {
+  it('should construct static fromOptions', () => {
+    const hardware = HardwareWallet.fromOptions('ledger');
+    assert.equal(hardware instanceof HardwareWallet, true);
+  });
+});
+
+// run certain tests only when device plugged in
+(async () => {
+  const hardware = HardwareWallet.fromOptions('ledger');
+  const devices = await hardware.getDevices();
+
+  if (devices.length > 0) {
+    describe('ledger hardware required tests', async () => {
+      let hardware;
+      const path = "m'/44'/0'/0'/0/0";
+
+      beforeEach(() => {
+        hardware = HardwareWallet.fromOptions('ledger');
+      });
+
+      afterEach(async () => {
+        await hardware.refresh();
+        hardware = null;
+      });
+
+      it('should get a real ledger device', async () => {
+        const devices = await hardware.getDevices();
+
+        assert.equal(Array.isArray(devices), true);
+        assert.equal(devices.length > 0, true);
```

Comment:
Doesn't really prove these are real devices does it? Isn't there some parent class we could compare against?

Reply:
I need to write a mock to return fake devices, I have mock logic in my other PR